radicle-node/src/test/tests.rs

```diff
@@ -34,6 +34,54 @@ use crate::{client, git, identity, rad, service, test};
 //
 // You may then run the test with eg. `cargo test -- --nocapture` to always show output.
 
+#[test]
+fn test_ping_pong() {
+    let mut alice = Peer::new("alice", [8, 8, 8, 8], MockStorage::empty());
+    let mut bob = Peer::new("bob", [9, 9, 9, 9], MockStorage::empty());
+
+    let mut sim = Simulation::new(
+        LocalTime::now(),
+        alice.rng.clone(),
+        simulator::Options::default(),
+    )
+    .initialize([&mut alice, &mut bob]);
+
+    alice.command(service::Command::Connect(bob.addr()));
+    sim.run_while([&mut alice, &mut bob], |s| !s.is_settled());
+
+    alice.ping(&bob.addr().ip()).unwrap();
+    sim.run_while([&mut alice, &mut bob], |s| !s.is_settled());
+
+    assert_eq!(PingState::Ok, alice.ping_resp(&bob.addr()).unwrap());
+}
+
+#[test]
+fn test_disconnecting_unresponsive_peers() {
+    let mut alice = Peer::new("alice", [8, 8, 8, 8], MockStorage::empty());
+    let mut bob = Peer::new("bob", [9, 9, 9, 9], MockStorage::empty());
+
+    let mut sim = Simulation::new(
+        LocalTime::now(),
+        alice.rng.clone(),
+        simulator::Options::default(),
+    )
+    .initialize([&mut alice, &mut bob]);
+
+    alice.command(service::Command::Connect(bob.addr()));
+    sim.run_while([&mut alice, &mut bob], |s| !s.is_settled());
+    assert_eq!(1, alice.sessions().negotiated().count(), "bob connects");
+
+    alice.elapse(IDLE_INTERVAL);
+    assert_eq!(
+        1, /* number of disconnects */
+        alice
+            .outbox()
+            .filter(|p| { matches!(p, Io::Disconnect(_, _)) })
+            .count(),
+        "disconnect unresponsive peer bob"
+    )
+}
+
```

Comment:
We should also test that our service responds to `ping` with a valid `pong`, and that invalid pongs are ignored.